### PR TITLE
5 skip loading pickle cache

### LINKED ISSUE #5: Caching is too sensitive.
PickleCache takes into account all variable in the chain members. It would be good to be able to specify explicitly which members it should ignore.
```python
class MyWorker:
    PickleCacheIgnore = ['workers']

    def __init__(self, num_examples=100, workers=10):
        self.num_examples = num_examples
        self.workers = workers
```

Second thing is to be able to deal with these cases:

```python
my_chain = Chain([
    PickleCache("cached_pkl", "id", my_chain_1),
    PickleCache("cached_pkl", "id", my_chain_2),
    PickleCache("cached_pkl", "id", my_chain_3),
])
```
if there is data cached for my_chain_3, we should be able to skip loading the pickles for my_chain_1 and my_chain_2. But only in the case, when cached my_chain_3 came from the same settings of my_chain_1 and my_chain_2.

One approach would be to update data['id'] after each PickleCache or even after each module so that the data['id'] would contain all the provenance of the data transformations it went through.

What do you think?

### PR DESCRIPTION
**Pull request by shakukse**
close #5

### CODE REVIEW

File: flexp/flow/README.md [hunk 1]
```diff
@@ -128,3 +128,42 @@ Solution for this is to store only the name of the database and open it in the `
...
+        self.attr1 = attr1
+        self.attr2 = attr2
+
+    def process(self, data):
+        pass
+
+my_chain = CachingChain([
+    # PickleCache ALWAYS change data key
+    PickleCache("cached_pkl", "id", [TestModule(12, 14, 18)]),  # key update
+    PickleCache("cached_pkl", "id", [TestModule(12, 10, 18)]),  # key update
```
Comment (alaneckhardt) [inline flexp/flow/README.md:179]: I would add one line with 
`PickleCache("cached_pkl", "id", [EmptyModule(1, 0)]),  # key update`

to show that PickleCache updates even when UpdateDataId is not specified.

File: flexp/flow/caching_chain.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    def _add(self, module):
+        """Add module to the chain
+        Check if a new module conforms to our requirements.
+
+        :param object|function module:
+        """
+        super()._add(module)
+        if isinstance(module, PickleCache):
+            # no need to distinguish PickleCache(chain=[Module1, Module2]) and [Module1, Module2]
+            # PickleCache.chain_info['chain_hash'] created same way: hash_dump_string(object_dump_to_string([module]))
```
Comment (alaneckhardt) [inline flexp/flow/caching_chain.py:53]: That's actually really good idea! I like that.

**Comment by alaneckhardt** [comment]
If you could please add a line to the example with EmptyModule and to refactor "|" separator to name attribute. Other than that it looks really nice to me. 
Looking forward using CachingChain instead of PickleCache!

File: flexp/flow/README.md (hunk 1 above)
Comment (shakukse) [reply flexp/flow/README.md:179]: Added

**Pull request merged**